File: src/main/java/com/youngjun/auth/storage/db/core/token/TokenEntity.java
package com.youngjun.auth.storage.db.core.token;

import com.youngjun.auth.core.domain.token.Token;
import com.youngjun.auth.storage.db.core.support.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;

@Table(name = "token")
@Entity
public class TokenEntity extends BaseEntity {

    private Long authId;

    private String refreshToken;

    protected TokenEntity() {
    }

    public TokenEntity(Long authId, String refreshToken) {
        this.authId = authId;
        this.refreshToken = refreshToken;
    }

    public Token toToken() {
        return new Token(authId, refreshToken);
    }

    public String getRefreshToken() {
        return refreshToken;
    }
}

File: src/main/kotlin/com/youngjun/auth/storage/db/core/token/TokenEntity.kt
package com.youngjun.auth.storage.db.core.token

import com.youngjun.auth.core.domain.token.RefreshToken
import com.youngjun.auth.core.domain.token.Token
import com.youngjun.auth.storage.db.core.support.BaseEntity
import jakarta.persistence.Entity
import jakarta.persistence.Table

@Table(name = "token")
@Entity
class TokenEntity(
    val authId: Long,
    val refreshToken: String,
) : BaseEntity() {
    fun toToken(): Token = Token(authId, RefreshToken(refreshToken))
}
